internal/restorer/restorer_unix_test.go

```diff
@@ -59,3 +60,97 @@ func TestRestorerRestoreEmptyHardlinkedFileds(t *testing.T) {
 		rtest.Equals(t, s1.Ino, s2.Ino)
 	}
 }
+
+func checkConsistentInfo(t testing.TB, file string, fi os.FileInfo, modtime time.Time, mode os.FileMode) {
+	if fi.Mode() != mode {
+		t.Errorf("checking %q, Mode() returned wrong value, want 0%o, got 0%o", file, mode, fi.Mode())
+	}
+
+	if !fi.ModTime().Equal(modtime) {
+		t.Errorf("checking %s, ModTime() returned wrong value, want %v, got %v", file, modtime, fi.ModTime())
+	}
+}
+
+// test inspired from test case https://github.com/restic/restic/issues/1212
+func TestRestorerConsistentTimestampsAndPermissions(t *testing.T) {
+	timeForTest := time.Date(2019, time.January, 9, 1, 46, 40, 0, time.UTC)
+
+	repo, cleanup := repository.TestRepository(t)
+	defer cleanup()
+
+	_, id := saveSnapshot(t, repo, Snapshot{
+		Nodes: map[string]Node{
+			"dir": Dir{
+				Mode:    0750 | os.ModeDir,
```

Comment:
Please create a wrapper function e.g. `normalizeFileMode` which on Windows returns `0555` for directories (`os.ModeDir`) and `0444` otherwise. Then also use it in `checkConsistentInfo`. I'm a bit surprised that filemode lies 0700 shouldn't work on Windows, at least NTFS is perfectly capable of emulating the standard unix permissions.

Reply:
Done, please tell me if that's what you had in mind.